docs/src/main/sphinx/admin/fault-tolerant-execution.rst

```diff
@@ -0,0 +1,344 @@
+       split weight-adjusted and serves as protection against situations where
+       catalogs report an incorrect split weight.
+
+       May be overridden for the current session with the
+       ``fault_tolerant_execution_min_task_split_count``
+       :ref:`session property <session-properties-definition>`.
+     - ``16``
+   * - ``fault-tolerant-execution-max-task-split-count``
+     - Maximum number of splits processed by single task. This value is not
+       split weight-adjusted and serves as protection against situations where
+       catalogs report an incorrect split weight.
+
+       May be overridden for the current session with the
+       ``fault_tolerant_execution_max_task_split_count``
+       :ref:`session property <session-properties-definition>`.
+     - ``256``
+
+Node allocation
+^^^^^^^^^^^^^^^
+
+With a task retry policy, nodes are allocated to tasks based on available memory
+and estimated memory usage. If task failure occurs due to exceeding available
+memory on a node, the task is restarted with a request to allocate the full node
+for its execution.
+
+The initial task memory-requirements estimation is static and configured with
+the ``fault-tolerant-task-memory`` configuration property. This property only
+applies to a ``TASK`` retry policy.
+
+.. list-table:: Node allocation configuration properties
+   :widths: 30, 40, 30
+   :header-rows: 1
+
+   * - Property name
+     - Description
+     - Default value
+   * - ``fault-tolerant-execution-task-memory``
+     - Initial task memory estimation used for bin-packing when allocating nodes
+       for tasks. May be overridden for the current session with the
+       ``fault_tolerant_execution_task_memory``
+       :ref:`session property <session-properties-definition>`.
+     - ``4GB``
+
+Other tuning
+^^^^^^^^^^^^
+
+The following additional configuration properties can be used to manage
+fault-tolerant execution:
+
+.. list-table:: Other fault-tolerant execution configuration properties
+   :widths: 30, 40, 30, 30
+   :header-rows: 1
+
+   * - Property name
+     - Description
+     - Default value
+     - Retry policy
+   * - ``fault-tolerant-execution-task-descriptor-storage-max-memory``
+     - Maximum amount of memory to be used to store task descriptors for fault
+       tolerant queries on coordinator. Extra memory is needed to be able to
+       reschedule tasks in case of a failure.
+     - (JVM heap size * 0.15)
+     - Only ``TASK``
+
+.. _fte-exchange-manager:
+
+Exchange manager
+----------------
+
+Exchange spooling is responsible for storing and managing spooled data for
+fault-tolerant execution. You can configure a filesystem-based exchange manager
+that stores spooled data in a specified location, either an S3-compatible
+storage system or a local filesystem.
+
+To configure an exchange manager, create a new
+``etc/exchange-manager.properties`` configuration file on the coordinator. In
```

Comment:
Does the config file need to be on all nodes or just the coordinator?

Reply:
All nodes.

Reply:
Got it, thanks. Clarified.